### ISSUE TITLE
random stuff imports marmalade

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10572]
|<img alt="marienz's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| marienz reported|
|-|-|
|Trac ID|trac#876|
|Type|defect|
|Created|2005-02-01 06:11:59Z|
|Branch|https://github.com/twisted/twisted/tree/delete-marmalade-876|

<details><summary>Searchable metadata</summary>

```
trac-id__876 876
type__defect defect
reporter__marienz marienz
priority__high high
milestone__ 
branch__branches_delete_marmalade_876 branches/delete-marmalade-876
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1107238319000000 1107238319000000
changetime__1229296154000000 1229296154000000
version__ 
owner__ 
cc__radix cc__itamarst cc__jml cc__jknight cc__marienz
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="marienz's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|marienz commented|
|-|-|

```
#!html
<pre>
If I install just the twisted-2.0.0a1 tarball, not the rest of twisted, and run
the unit tests, a bunch of tests fail because they use marmalade, which uses
twisted.web.microdom, which isn't there:

- t.test.test_reflector because it imports twisted.enterprise.xmlreflector
- t.test.test_sob.PersistTestCase
- t.test.test_import.AtLeastImportTestCase, again via enterprise.xmlreflector
- t.test.test_application.TestLoading
- t.test.test_application.TestAppSupport
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

```
#!html
<pre>
We should have a story for this. My opinion: deprecate marmalade and 
xmlreflector to death, and say any problems with it are too bad so sad stop 
using it already. And make the tests skip, too, ideally.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
Crap, this needs to be fixed by release.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

```
#!html
<pre>
So tasks are (just making certain):
- Deprecate marmalade
- Deprecate xmlreflector
- Skip tests
- Make sure not used elsewhere
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

```
#!html
<pre>
Deprecate == Deprecate with warning
Skip tests only when t.web not present
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

```
#!html
<pre>
The test_application tests that get skipped don't really need to depend on 
twisted.web. They should be replaced, but that's not for2.0 I don't think.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Not urgent.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml removed owner|
|-|-|

I'm not ever going to do this.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun commented|
|-|-|

(In [25576]) Branching to 'delete-marmalade-876'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_12"></a>|@therve set owner to @exarkun|
|-|-|

`+           for t in ['python',  'source', 'file'] if config[t]][0]`

There is one extra space before 'source'.

One flake:
`twisted/test/test_application.py:4: 'warnings' imported but unused`

Please merge!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun commented|
|-|-|

(In [25665]) address review feedback


fix whitespace in app.py and fix imports/copyright statement/module docstring for app tests

refs [#10572](https://github.com/twisted/twisted/issues/10572)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun set status to `closed`|
|-|-|

(In [25666]) Merge delete-marmalade-876

Author: exarkun, itamarst
Reviewer: therve
Fixes: [#10572](https://github.com/twisted/twisted/issues/10572)

Delete `twisted.persisted.marmalade`, a long-deprecated XML Python Object
serialization library, and all of the vestigal code and documentation in
Twisted which still referred to it.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
